binary/Pos/Binary/Class/TH.hs

```diff
@@ -175,10 +184,6 @@ deriveSimpleBiInternal predsMB headTy constrs = do
     tagType :: TypeQ
     tagType = [t| Word8 |]
 
-    -- Helpers --
-    failText :: MonadFail m => T.Text -> m a
-    failText = fail . toString
```

Comment:
Now there are 100500 usages of `toTemplateHaskellError . Left`. Why not leave `(failText :: Text -> Q a)  = toTemplateHaskellError . Left`? Maybe `failText` is not the best name, but other than that such function makes perfect sense to me 🤷‍♂️

Reply:
Primarily for consistency, but I see your point, I'll restore this function under a different name. I don't like the name `failText` because it makes harder to grep for `fail`, thus to adhere to the policy of no `MonadFail`.

Reply:
I replaced it with `templateHaskellError`.